subprojects/docs/src/docs/userguide/taskConfiguration.adoc

```diff
@@ -0,0 +1,269 @@
+
+[[sec:old_vs_new_configuration_api_overview]]
+=== Old vs New API overview
+
+[NOTE]
+====
+Methods that take a `groovy.lang.Closure` are covered in the new API with methods taking `org.gradle.api.Action`.
+====
+
+[cols="a,a", options="header"]
+|===
+| Old API | New API
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.util.Map)[]
+| No direct equivalent.
+2+| Use one of the alternatives below.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.util.Map,groovy.lang.Closure)[]
+| No direct equivalent.
+2+| Use one of the alternatives below.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String,groovy.lang.Closure)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String,org.gradle.api.Action)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String,java.lang.Class)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String,java.lang.Class)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String,java.lang.Class,org.gradle.api.Action)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String,java.lang.Class,org.gradle.api.Action)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskContainer#create(java.lang.String,java.lang.Class,java.lang.Object...)[]
+| api:org.gradle.api.tasks.TaskContainer#register(java.lang.String,java.lang.Class,java.lang.Object...)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskCollection#getByName(java.lang.String)[]
+| api:org.gradle.api.tasks.TaskCollection#named(java.lang.String)[]
+2+| This returns a `TaskProvider` instead of a `Task`.
+
+| api:org.gradle.api.tasks.TaskCollection#getByName(java.lang.String,groovy.lang.Closure)[]
+| `named(java.lang.String).configure(Action)`
+2+| This returns `void`, so it cannot be chained.
+
+| api:org.gradle.api.tasks.TaskContainer#getByPath(java.lang.String)[]
+| No direct equivalent.
+2+| Accessing tasks from another project requires a specific ordering of project evaluation. 
+
+| api:org.gradle.api.NamedDomainObjectCollection#findByName(java.lang.String)[]
+| No direct equivalent.
+2+| `named(String)` is the closest equivalent, but will fail if the task does not exist. Using `findByName(String)` will cause lazy tasks to be created/configured.
+
+| api:org.gradle.api.tasks.TaskContainer#findByPath(java.lang.String)[]
+| No direct equivalent.
+2+| See `getByPath(String)` above.
+
+| api:org.gradle.api.tasks.TaskCollection#withType(java.lang.Class)[]
+| _OK_
+2+| This is OK to use because it does not require tasks to be created immediately.
```

Comment:
It would be good to mention that iterating over this collection will resolve all the task provider in this collection, so it's something to avoid as well. Prefer `withType(...).configureEach`.